src/OpenLoco/Widget.cpp

```diff
@@ -226,61 +226,60 @@ namespace OpenLoco::Ui
 
     void Widget::sub_4CADE8(Gfx::Context* context, const Window* window, AdvancedColour colour, bool enabled, bool disabled, bool activated)
     {
-        int16_t xPlaceForImage = left + window->x;
-        int16_t yPlaceForImage = top + window->y;
-        uint32_t imageId = image;
+        Ui::Point placeForImage(left + window->x, top + window->y);
+        const bool isColourSet = image & Widget::imageIdColourSet;
+        ImageId imageId = ImageId::fromUInt32(image & ~Widget::imageIdColourSet);
         if (type == WidgetType::wt_6 || type == WidgetType::toolbarTab || type == WidgetType::tab || type == WidgetType::wt_4)
         {
             if (activated)
             {
                 // TODO: remove image addition
```

Comment:
does this change mean we can remove this comment?

Reply:
No, tbh I'm not entirely sure how we will resolve this. The comment is there because we really should have some safe way of to define a widgets different image states but it's still far off from possible.

Reply:
no worries, if its future work then leaving the comment is fine